Sources/AppStoreConnectCLI/Services/EndpointRequestor.swift

```diff
@@ -24,3 +24,28 @@ struct DefaultEndpointRequestor: EndpointRequestor {
         }
     }
 }
+
+protocol PaginatedResponse: Decodable {
+    var links: PagedDocumentLinks { get }
+}
+
+extension EndpointRequestor {
+    func concatFetcher<T: PaginatedResponse>(
+        with endpointMaker: @escaping (PagedDocumentLinks?) -> APIEndpoint<T>,
+        next: PagedDocumentLinks?
+    ) -> AnyPublisher<[T], Error> {
+        self.request(endpointMaker(next)).flatMap { (response) -> AnyPublisher<[T], Error> in
+            if response.links.next != nil {
+                return self.concatFetcher(with: endpointMaker, next: response.links)
+                    .flatMap {
+                        Empty<[T], Error>()
+                            .append([response] + $0)
+                    }
+                    .eraseToAnyPublisher()
+            }
+
+            return Empty<[T], Error>().append([response]).eraseToAnyPublisher()
+        }
+        .eraseToAnyPublisher()
+    }
+}
```

Comment:
🤔 I think we can make a couple small improvements here
What do you think of the function name: `requestAllPages`

Additionally we can default `next` to nil if not provided (next: PagedDocumentLinks? = nil)

Rather than the `Maker` suffix we usually see the `Provider` suffix. Or we might even bet able to get away with just naming it `endpoint` as a function

Also if we flip the parameters to provide the closure second we can have the option to use trailing closure syntax